src/util/orm.py

```diff
@@ -288,25 +288,30 @@ def recompute_fields(cr, model, fields, ids=None, logger=_logger, chunk_size=256
     model = Model._name
 
     if ids is None:
-        query = format_query(cr, "SELECT id FROM {} ORDER BY id", table_of_model(cr, model)) if query is None else query
+        query = format_query(
+            cr,
+            "WITH query AS ({}) SELECT * INTO UNLOGGED _upgrade_rf FROM query",
+            SQLStr(
+                format_query(cr, "SELECT id FROM {} ORDER BY id", table_of_model(cr, model)) if query is None else query
+            ),
+        )
         cr.execute(query)
-        count = cr.rowcount
-        if count < 2**21:  # avoid the overhead of a named cursor unless we have at least two chunks
-            ids_ = (id_ for (id_,) in cr.fetchall())
-        else:
+        cr.execute("SELECT count(*) FROM _upgrade_rf")
+        count = cr.fetchone()[0]
```

Comment:
Can't we get back to `rowcount` if we use CREATE TABLE AS? Is there any special case that would break it? I don't see any.
```suggestion
        count = cr.rowcount
```